JDP190303-25 implementation UserController, added UserMapper and UserService tests

karolpolikarp opened the pull request:
- this implemenation lacks testGetUser()

Commit:
Expanding implementation of UserController, added UserMapper and UserService tests

Commit:
Corrected testGetUser

Commit:
Problem with gradlew

Commit:
Changed application.properties

Commit:
Fighting with ./gradlew build

Commit:
fighting with gradlew

Commit:
testGetUser

Commit:
Changes to UserController, added UserMapper and UserService tests

fmatusiak reviewed (commented):
Hej hej , hmm w tescie list masz 
Assert.assertEquals(usersList.get(1).getUserName(), testUsersList.get(1).getUserName()); czyli tak naprawde sprawdzenie pierwszego Usera z listy a nie wszystkich ,wiec nie wiem czy nie powinno byc sprawdzania wszytkich userow ,a tak to jest ok :D . Pozdrawiam

interceptor2048 commented on src/main/java/com/kodilla/ecommercee/controller/UserController.java:
```diff
@@ -37,12 +35,11 @@ public void createUser(@RequestBody UserDto userDto) {
 
     @PutMapping("blockUser")
     public UserDto blockUser(@RequestParam Long userId, @RequestBody UserDto userDto) {
```
Potrzebne jest całe Dto? Nie wystarczy przekazać userId i na tej podstawie wyciągnąć w service resztę?

rbarcik commented:
@PostMapping("createUser") dodałbym consumes = APPLICATION_JSON_VALUE
oraz import static org.springframework.util.MimeTypeUtils.APPLICATION_JSON_VALUE;

w @PutMapping("blockUser") 
return new UserDto(userId, userDto.getUserName(), true, userDto.getUserIdKey()); 
zwraca odpowiedź ale nie zapisuje jej do bazy

zrobiłbym jeszcze testy do mapToUser, a zwłaszcza pod kątem userId, według mnie ta metoda nie mapuje userId,

rbarcik reviewed (commented):
@PostMapping("createUser") dodałbym consumes = APPLICATION_JSON_VALUE
oraz import static org.springframework.util.MimeTypeUtils.APPLICATION_JSON_VALUE;

w @PutMapping("blockUser") 
return new UserDto(userId, userDto.getUserName(), true, userDto.getUserIdKey()); 
zwraca odpowiedź ale nie zapisuje jej do bazy

zrobiłbym jeszcze testy do mapToUser, a zwłaszcza pod kątem userId, według mnie ta metoda nie mapuje userId,

Commit:
testv2

Commit:
v1

Commit:
string

Commit:
Correcting User

karolpolikarp commented:
Correcting previous mistakes, applying reviewed suggestions.
- made corrections to User entity
- fixed blockUser in UserController, now works fine and saves data in repository
- added more tests to getUsersList
- added consumes to createUser in UserController
- fixed UserMapper in terms of userId

rbarcik commented on src/main/java/com/kodilla/ecommercee/controller/UserController.java:
```diff
@@ -37,11 +37,8 @@ public void createUser(@RequestBody UserDto userDto) {
-        userToBlockDto.setIsBlocked(true);
-        User userToBlock = userMapper.mapToUser(userToBlockDto);
-        userService.saveUser(userToBlock);
+    public User blockUser(@RequestParam Long userId) throws UserNotFoundException {
```
Nie wiem czy to ma znaczenie skoro działa ale pomyśl o tym co pisał Andrzej 
"Potrzebne jest całe Dto? Nie wystarczy przekazać userId i na tej podstawie wyciągnąć w service resztę?"

karolpolikarp replied:
Hej, własnie wydaje mi się, że to zaimplementowałem, skoro metoda nie posiada teraz @RequestBody UserDto tylko na podstawie podania tylko userId zmienia wyłącznie jego wartość isBlockec

rbarcik replied:
Dobra, zatwierdzimy to jak jest, żeby zmergować pullrequest, ewentualnie na spotkaniu zapytamy Andrzeja.

rbarcik reviewed (approved):
Nie wiem czy to ma znaczenie skoro działa ale pomyśl o tym co pisał Andrzej
"Potrzebne jest całe Dto? Nie wystarczy przekazać userId i na tej podstawie wyciągnąć w service resztę?"
